src/jit/regset.cpp

```diff
@@ -3458,19 +3458,52 @@ bool                genIsProperRegPair(regPairNo regPair)
  *
  *  Given a register that is an argument register 
  *   returns the next argument register 
+ *
+ *  Note: that this method will return a non arg register 
+ *   when given REG_ARG_LAST
+ *
```

Comment:
It more that I am documenting the behavior not choosing it.

I tried adding an assert that  we were returning a valid argument register, but that failed.
I probably could also change it to return REG_NA and that would probably work.